includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -835,6 +839,30 @@ public function clear_upe_appearance_transient() {
 		delete_transient( self::UPE_APPEARANCE_TRANSIENT );
 	}
 
+	/**
+	 * Sets the title on checkout correctly before the title is displayed.
+	 *
+	 * @param {string} $title The title of the gateway being filtered.
+	 * @param {string} $id    The id of the gateway being filtered.
+	 *
+	 * @return string Filtered gateway title.
+	 */
+	public function maybe_filter_gateway_title( $title, $id ) {
+		if ( self::GATEWAY_ID === $id && $this->title === $title ) {
```

Comment:
This check for `$this->title === $title` is probably a little redundant, but eh, better safe than sorry, I suppose!

Reply:
Ya, I thought about others that might be using the filter, or if the string is translated, etc, and I didn't want to override an override.